### PR TITLE
Waqfeya booksearching

### PR DESCRIPTION
**Pull request by 3akev**

### CODE REVIEW

**Comment by martomato** [comment]
I should be able to review this within 48 hours

File: tests/extensions/booksearch/test_booksite.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+@pytest.mark.asyncio
+async def test_convert_raises_error_if_not_supported_site(ctx):
+    site = 'nonexistentlibrary'
+
+    with pytest.raises(UnSupportedSite):
+        await BookSite.convert(MagicMock(spec=ctx), site)
+
+
+@pytest.mark.asyncio
```
Comment (martomato) [inline tests/extensions/booksearch/test_booksite.py:32]: comment above this method(?)/somewhere in this file detailing Unicode information and charset settings? I remember there was some unicode wrestling, would be good to have documentation for future reference here :)

**Review by martomato — CHANGES_REQUESTED** [review]
code comments are good, esp for niche obstacles

**Review by martomato — APPROVED** [review]
JAK

**Comment by 3akev** [comment]
Wa iyyak. Merge please?

**Pull request merged**